WooCommerce/Classes/Tools/InfiniteScroll/ScrollWatcher.swift

```diff
@@ -0,0 +1,43 @@
+import UIKit
+
+/// Watches for scroll content offset changes and notifies its subscribers if the scroll position is over a threshold.
+///
+final class ScrollWatcher {
```

Comment:
I'm thinking if it would be better to call it `ScrollObserver`, but maybe it's confusing? I don't know, I don't have a strong opinion about this name 👍 feel free to leave it 😉

Reply:
*Observer might sound like the Observer in Rx, especially now that we have some lightweight Rx code like [`Observer`](https://github.com/woocommerce/woocommerce-ios/blob/74332bed838a5f4312586c450bcb16d2d1cc69ff/WooCommerce/Classes/Tools/Observables/Observer.swift) in the code base 🤔 keeping this name `ScrollWatcher` for now but lemme know if you have other suggestions!